### PR TITLE
feat: VoIP freeswitch UI admin

### PR DESCRIPTION
**Pull request by aleksandernsilva**
## Proposed changes (including videos or screenshots)

This PR adds all necessary logic for assignment and unasignment of VoIP extensions in the admin users page. Configuring this information will allow users to make and receive call in team collab. 

#### Assign/Remove extension button added to the workspace user list
Screenshot 2024-08-15 at 11 10 07

#### Assign extension modal
Screenshot 2024-08-15 at 11 10 30

#### Remove extension modal
Screenshot 2024-08-15 at 11 10 47

## Issue(s)
[VOIP-71](https://rocketchat.atlassian.net/browse/VOIP-71)

## Steps to test or reproduce
- Run your workspace
- Enable "VoIP for Team Collaboration" in the workspace settings
- Go to `/admin/users`

## Further comments
Backend can be found in this PR: #32435

[VOIP-71]: https://rocketchat.atlassian.net/browse/VOIP-71?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/views/admin/users/UsersTable/UsersTable.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { mockAppRoot } from '@rocket.chat/mock-providers';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import UsersTable from './UsersTable';
+
+jest.mock('../../../../../app/models/client/models/Roles', () => ({ Roles: { findOne: () => undefined } }));
+jest.mock('../../../../components/UserStatus', () => ({ UserStatus: () => null }));
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/users/UsersTable/UsersTable.spec.tsx:9]: Why is this needed?

File: apps/meteor/client/views/admin/users/voip/AssignExtensionModal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+				statusText: '',
+				avatarETag: null,
+			} as any,
+		],
+		success: true,
+	}));
+
+// TODO: it('should load with default user', async () => {});
+
+// TODO: it('should load with default extension', async () => {});
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/users/voip/AssignExtensionModal.spec.tsx:47]:
```suggestion
it.todo('should load with default user', async () => {});

it.todo('should load with default extension', async () => {});
```

File: apps/meteor/client/views/admin/users/voip/AssignExtensionModal.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+								name='extension'
+								render={({ field }) => (
+									<Select
+										id={freeExtensionNumberId}
+										disabled={isLoading || !selectedUsername}
+										options={extensionOptions}
+										placeholder={t('Select_an_option')}
+										value={field.value}
+										onChange={field.onChange}
+										data-testid='input-free-extension-numbers'
```
Comment (tassoevan) [inline apps/meteor/client/views/admin/users/voip/AssignExtensionModal.tsx:130]: Why it cannot be queried by role?

**Review by tassoevan — CHANGES_REQUESTED** [review]

**Review by MartinSchoeler — APPROVED** [review]

**Review by pierre-lehnen-rc — APPROVED** [review]
backend

**Pull request merged**